[java] Parser error with private method in nested classes in interfaces

adangel opened the pull request:
Fixes #793

jsotuyod commented:
@adangel if the flag is reset after finding a nested enum / class / anonymous class, doesn't is miss any following up private methods?

ie:
```java
public interface InterfaceWithInnerClass {

    Object FOO = new Object() {
        private void privateMethod() { }
    };

    class InnerClass {
        private void privateMethod() { }
    }
 
     enum InnerEnum {
         VALUE;
         private void privateMethod() { }
     }

    private void foo(); // bad, but not reported as the flag has been reset 3 times already
 }
```

It seems like we need to have a stack of flags, pushing a new value each time we enter a new class / interface / annotation / enum / anonymous class; and poping as we leave... are my assumptions correct?

adangel commented:
@jsotuyod You are right - nesting wasn't supported. I've fixed the issue and added an additional test case. I didn't yet use `Stack<Boolean>`, but just remembered the old value in the current method stack the parser is executing.
I can rewrite the logic using a separate stack, if this seems better.

vanniktech commented:
Whops sorry for that. I messed up my tabs.

jsotuyod commented:
@adangel amazing, it looks good now. I'll do some manual testing myself and merge if I find no further issues.
